pkg/scheduler/scheduler.go

```diff
@@ -589,14 +589,10 @@ func (s *Scheduler) getInitialAssignments(log logr.Logger, wl *workload.Info, sn
 func (s *Scheduler) evictWorkloadAfterFailedTASReplacement(ctx context.Context, log logr.Logger, wl *kueue.Workload) error {
 	log.V(3).Info("Evicting workload after failed try to find a node replacement; TASFailedNodeReplacementFailFast enabled")
 	msg := fmt.Sprintf("Workload was evicted as there was no replacement for a failed node: %s", wl.Status.UnhealthyNodes[0].Name)
-	err := workload.Evict(
+	return workload.Evict(
 		ctx, s.client, s.recorder, wl, kueue.WorkloadEvictedDueToNodeFailures, msg, "", s.clock,
 		workload.EvictWithLooseOnApply(), workload.EvictWithRetryOnConflictForPatch(),
 	)
-	if err != nil {
-		return err
-	}
-	return nil
```

Comment:
Let's avoid the drive-by changes which are distraction here and can make it more work to cherry-pick.

Reply:
Reverted and prepared cleanup https://github.com/kubernetes-sigs/kueue/pull/8285.